src/Storages/MergeTree/checkDataPart.cpp

```diff
@@ -377,7 +377,16 @@ IMergeTreeDataPart::Checksums checkDataPart(
             auto file_name = it->name();
             if (!data_part_storage.isDirectory(file_name))
             {
-                auto remote_path = data_part_storage.getRemotePath(file_name);
+                const bool is_projection_part = data_part->isProjectionPart();
+                auto remote_path = data_part_storage.getRemotePath(file_name, /* if_exists */is_projection_part);
```

Comment:
so we moved throwing error for missing projection part here for better error message is it ?

Reply:
Not exactly - to avoid a logical error, because in case of broken projection there will be no path here, but `getRemotePath` used to throw a logical error in this case, which was caught by stress test when running the test I added in this PR. A logical error in stress test is std::terminate.